### PR TITLE
PTX-15169 [TC Automation] DriveAddMetaDataDiskStatusCheck

### PR DESCRIPTION
**Pull request by spai-px**
What this PR does / why we need it:
Adds test case DriveAddMetaDataDiskStatusCheck
Which issue(s) this PR fixes (optional)
Closes # PTX-15169

Special notes for your reviewer:
jenkins : [jenkins link](https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-pxe-byoc/2034/consoleFull)
Aetos: [aetos](https://aetos.pwx.purestorage.com/resultSet/testSetID/581752)

### CODE REVIEW

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -11356,3 +11356,184 @@ var _ = Describe("{driveAddPxRestart}", func() {
...
+							log.Infof("Cannot add metadata device when internal kvdb is running on this node")
+							err = nil
+							continue
+						}
+					}
+					log.FailOnError(err, "Failed to add metadata device on node : %s", selectedNode.Name)
+
+					// exit pool maintenance
+					err = Inst().V.ExitPoolMaintenance(storageNode)
+					log.FailOnError(err, "Node: %v Failed to exit out of maintenance mode", storageNode.Name)
```
Comment (kphalgun-px) [inline tests/basic/storage_pool_test.go:11430]: How about we defer `ExitPoolMaintenance`?

Comment (kphalgun-px) [reply tests/basic/storage_pool_test.go:11430]:
```go
err = Inst().V.EnterPoolMaintenance(storageNode)
log.FailOnError(err, "node: %v failed to transition to pool maintenance mode", storageNode.Name)
defer func() {
    err = Inst().V.ExitPoolMaintenance(storageNode)
    log.FailOnError(err, "Node: %v Failed to exit out of maintenance mode", storageNode.Name)
}()
```

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -11356,3 +11356,184 @@ var _ = Describe("{driveAddPxRestart}", func() {
...
+
+					selectedNode = storageNode
+					break
+				}
+			}
+			//check if selecteNode is empty or not
+			if selectedNode.Name == "" {
+				log.FailOnError(fmt.Errorf("No node found with metadata disk or metadata disks cannot be added to any nodes"), "No node found with metadata disk ")
```
Comment (kphalgun-px) [inline tests/basic/storage_pool_test.go:11438]: No node is without* a metadata disk?

File: tests/basic/storage_pool_test.go (hunk 1 above)
Comment (spai-px) [reply tests/basic/storage_pool_test.go:11430]: Since the defer is inside the loop it won't run for every loop so this won't work.

**Pull request merged**